content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,103 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
 
 # Automatic Sharding
 
-> *Note:* Automatic sharding is not yet part of this specification.
-This section merely serves as an outlook.
-A specification of automatic sharding will be added to this document in a future version.
-
-Automatic sharding is a method for scaling Waku relay in the number of (smaller) content topics.
-It automatically maps Waku content topics to pubsub topics.
-Clients and protocols building on Waku relay only see content topics, while Waku relay internally manages the mapping.
-This provides both scaling as well as removes confusion about content and pubsub topics on the consumer side.
-
-From an app point of view, a subscription to a content topic `waku2/xxx` using automatic sharding would look like:
-
-`subscribe("/waku2/xxx", auto=true)`
-
-The app is oblivious to the pubsub topic layer.
-(Future versions could deprecate the default pubsub topic and remove the necessity for `auto=true`.)
-
-*The basic idea behind automatic sharding*:
-Content topics are mapped using [consistent hashing](https://en.wikipedia.org/wiki/Consistent_hashing).
-Like with DHTs, the hash space is split into parts,
-each covered by a Pubsub topic (mesh network) that carries content topics which are mapped into the respective part of the hash space.
+Autosharding selects shards automatically and is the default behavior for shard choice.
+The other choices being static and named sharding as seen in previous sections.
+Shards (pubsub topics) SHOULD be computed from content topics with the procedure below.
+
+## Rendezvous Hashing
+
+Also known as the Highest Random Weight (H.R.W.) method, has many properties useful for shard selection.
+- Distributed agreement. Local only computation without communication.
+- Load balancing. Content topics are spread as equally as possible to shards.
+- Minimal disruption. Content topics switch shards infrequently.
+- Low overhead. Easy to compute.
+
+### Algorithm
+
+For each shard,
+hash using Sha2-256 the concatenation of
+the content topic `application` field (N U.T.F.-8 bytes),
+`version` (N U.T.F.-8 bytes),
+the `cluster` index (2 bytes) and
```

Comment:
i see.
* why 2^16? isnt it too much. we should be able to count the networks with the fingers of our hands, so i would suggest 4 or 8 to be covered, but not 16.
* why 1024? afaik this would be the number of shards that we define in each generation? we were going to launch with 10, 20ish?

Reply:
@alrevuelta I agree we should change it but it's out of scope for this PR.